src/data_lakehouse_ingest/core.py

```diff
@@ -1,23 +1,138 @@
+"""
+Core orchestration module for the Data Lakehouse Ingest framework.
+Executes config-driven ingestion from Bronze (raw) to Silver (curated) Delta tables.
+Handles schema enforcement (SQL/LinkML), multi-format loading, and report generation.
+"""
+
 import json
-from typing import Union
+import logging
+from typing import Any
+from datetime import datetime, timezone
+from minio import Minio
+from pyspark.sql import SparkSession
+from pyspark.sql.utils import AnalysisException
+
+from .utils.report_utils import generate_report
+from .logger import safe_log_json
+from .config_loader import ConfigLoader
+
+# New modular helpers
+from .orchestrator.init_utils import init_logger, init_run_context
+from .orchestrator.table_processor import process_table
+from .orchestrator.error_utils import error_entry_for_exception
 
-def ingest_from_config(config: Union[str, dict], spark=None) -> bool:
+
+def ingest(
+    config: str | dict[str, Any],
+    spark: SparkSession | None = None,
+    logger: logging.Logger | None = None,
+    minio_client: Minio | None = None,
+) -> dict[str, Any]:
     """
-    Minimal ingestion stub. Returns True for both dict and JSON path configs.
+    Orchestrates the end-to-end data ingestion process defined by a configuration.
+
+    Loads raw (Bronze) data from local or S3/MinIO sources, applies schema enforcement
+    (SQL or LinkML-based), and writes curated (Silver) Delta tables. Generates a
+    structured report summarizing table-level outcomes and errors.
 
     Args:
-        config (Union[str, dict]): Either a dict config or a JSON file path.
-        spark (optional): SparkSession (not used in v0.0.1).
+        config (str | dict[str, Any]): Path to the config file (local or s3a://) or an inline config dictionary.
+        spark (SparkSession, optional): Active Spark session used for reading and writing data.
+        logger (logging.Logger, optional): Logger instance for structured logging.
+        minio_client (Minio, optional): MinIO client used to read configuration or data from S3-compatible sources.
 
     Returns:
-        bool: Always True
+        dict[str, Any]: A structured ingestion report containing status, errors, and table-level metrics.
+
+    Notes:
+        - SparkSession must be provided by the caller.
+        - Supports multiple file formats (CSV, TSV, JSON, XML).
+        - Each table in the configuration is processed independently.
     """
-    if isinstance(config, str):
-        # Treat as path to JSON file
-        with open(config, "r") as f:
-            _ = json.load(f)
-    elif isinstance(config, dict):
-        _ = config
-    else:
-        raise TypeError("config must be a dict or a JSON file path string")
-    return True
+    started_at = datetime.now(timezone.utc).isoformat()
```

Comment:
It's probably easier to pass around `datetime` objects internally (in case you have to do any calculations on them) and then convert them to isoformat when you output them. However, I don't have the full context for how `started_at` is used elsewhere in the codebase, so this may or may not be relevant advice.